File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        int n=text.size();

        unordered_map<char,int> have;           // The Hashmap of have declaration
        for(int i=0;i<n;i++){
            have[text[i]]++;
        }

        unordered_map<char,int>need;            // The Hashmap of the need Declaration
        need['b']=1;
        need['a']=1;
        need['l']=2;
        need['o']=2;
        need['n']=1;

        int res=INT_MAX;                        // The Minimum req will decide the ans

        for(auto i:need){
            char c=i.first;          //Key 
            int fneed=i.second;      //Value 
            int fhave=have[c];       // Have is being taken on from the have hashmap with the letters
            int times=fhave/fneed;   // Gives the times 

            res=min(res,times);      //The Min of the both will be the ans
        }
        return res;                  // The Min no. of the times {The Final Answer}
    }
};

File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        n=len(text)

        have={}
        for i in range(n):
            have[text[i]]=have.get(text[i], 0) + 1

        need={}
        need['b']=1
        need['a']=1
        need['l']=2
        need['o']=2
        need['n']=1

        res=float('inf')

        for c, fneed in need.items():
            fhave=have.get(c,0)
            times=fhave//fneed

            res=min(res,times)

        return res